docs-general/onboard.md

```diff
@@ -31,33 +31,54 @@ This is a rough depiction of how webinfra code runs in practice / in deployment.
+
+(_Google Cloud Storage_ is is Google's simple file storage, which works similarly to Google Drive or other file-and-folder based stores. At the top level, there are "buckets", which contain a filesystem with folders and files. Buckets are mostly the same as a folder, but they also specify properties (e.g. locations, public/private), which is useful for separating data based on those properties. `mitbattlecode-production-ephemeral` is a special bucket configured to auto delete files after one day, used for resume book temporary storage and other temp needs.)
+
+This request is then taken off the respective Pub/Sub collection and processed by either a _compile runner_ (also called a compile server or submission server) or a _execute runner_ (also called an execute server or scrimmage server). A compile runner takes in the request, gets the competitor's source code from Cloud Storage, compiles this code through running Java commands, uploads the compiled code back to Cloud Storage, and tells the backend to update the database accordingly (to reflect the completion of compilation). An execute runner follows a similar process, though with matches rather than submissions.
+
+#### Deploys
+
+Of course, all our machines (servers, databases, etc) must be deployed somewhere in order to run.
+
+Our backend is deployed via _Cloud Run_. This is more like a set of machines that Google operates, that runs code for you. Because the machines are operated by Google, we don't get much power over them. For example we can't SSH into them. Cloud Run may run multiple instances of your code (and will spread out the load). We use this because it's generally cheaper. But since it's limited for certain operations, we can't use Cloud Run for everything.
+
+Our compile and execute runners are deployed and run through Google Cloud's _Compute Engine_ which allows you to create and run arbitrary virtual machines (VMs). Our virtual machines are deployed via Docker, an system for constructing VMs with any content and running those VMs on a variety of operating systems. For example, we can use Docker to install Django and automatically run your code. Learn more at [their page](https://www.docker.com).
+
+Builds (the process of going from source code to things running on deployed machines) are done via _Cloud Build_. We configure Cloud Build to run in response to various "triggers", and we specify our triggers to be new tag created on a repository.
+
+To specify a commit to be deployed, on GitHub, assign a _tag_ to that commit. In particular, give the tag a certain prefix (see previous tags for examples). Cloud Build will see the tag, notice it matches a configured regex pattern, use the Docker information in your repo, and deploy what is specified by your tag!
+
+All our configuration about deploys is specified via _Terraform_, an infrastructure-as-code service. See the `deploy` file for more.
```

Comment:
ty! having deploy instructions is awesome

Reply:
yaaa it rlly is
you should also thank whoever wrote these -- ie you and jerry